### PR TITLE
planner: check plan type when executing `explain format="dot"`.

### LINKED ISSUE #17141: planner: refine the infomation when execute `desc format="dot"` for `core.Explain`.
### 1. What did you do?

```sql
desc select 1+1;
desc for connection 1; 
desc select 1+1;
desc format="dot" for connection 1;
```

### 2. What did you expect to see?

Output an empty graph, no error.

### 3. What did you see instead?

```sql
mysql> desc for connection 1;
Empty set (0.00 sec)

mysql> desc format="dot" for connection 1;
ERROR 1105 (HY000): interface conversion: *core.Explain is not core.PhysicalPlan: missing method Children
```

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)

```sql
mysql> select tidb_version();
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()

          |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-beta.2-441-g36a29da52
Git Commit Hash: 36a29da529c4abc1f59958a465469aca8726e492
Git Branch: master
UTC Build Time: 2020-05-12 11:51:52
GoVersion: go1.14.1
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by LittleFall**
### What problem does this PR solve?

Issue Number: close #17141 

### What is changed and how it works?

Check plan type when executing `explain format="dot"`.

### Check List 

Tests 

- Unit test

### Release note 

Check plan type when executing `explain format="dot"`.

### CODE REVIEW

File: executor/explainfor_test.go [hunk 1]
```diff
@@ -185,3 +185,21 @@ func (s *testPrepareSerialSuite) TestExplainForConnPlanCache(c *C) {
...
 	))
 }
+
+func (s *testPrepareSerialSuite) TestExplainDotForExplainPlan(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+
+	rows := tk.MustQuery("select connection_id()").Rows()
+	c.Assert(len(rows), Equals, 1)
+	connID := rows[0][0].(string)
+	tk.MustQuery("explain select 1").Check(testkit.Rows(
```
Comment (SunRunAway) [inline executor/explainfor_test.go:195]:
```suggestion
	tk.MustQuery("explain format="dot" select 1").Check(testkit.Rows(
```

Comment (LittleFall) [reply executor/explainfor_test.go]: great thanks for catch it 
😭

File: executor/explainfor_test.go [hunk 2]
```diff
@@ -185,3 +185,21 @@ func (s *testPrepareSerialSuite) TestExplainForConnPlanCache(c *C) {
...
+		"Projection_3 1.00 root  1->Column#1",
+		"└─TableDual_4 1.00 root  rows:1",
+	))
+
+	tkProcess := tk.Se.ShowProcess()
+	ps := []*util.ProcessInfo{tkProcess}
+	tk.Se.SetSessionManager(&mockSessionManager1{PS: ps})
+
+	tk.MustQuery(fmt.Sprintf("explain for connection %s", connID)).Check(nil)
```
Comment (SunRunAway) [inline executor/explainfor_test.go:204]: This line should be changed, too

Comment (LittleFall) [reply executor/explainfor_test.go]: Actually, in my original vision, the first one does not need to be added(it is just an explain), and this one needs to be added.

**Comment by LittleFall** [comment]
@zz-jason @SunRunAway Could you help me merge it? Thanks!

File: executor/explainfor_test.go (hunk 1 above)
Comment (zz-jason) [reply executor/explainfor_test.go:195]: it seems that the test is unchanged.

Comment (LittleFall) [reply executor/explainfor_test.go:195]: > it seems that the test is unchanged.

Yes, actually, in my original vision, the first one does not need to be added(it is just an explain), and the second one needs to be added.

So I just fix the second explain.

**Pull request merged**